app/models/space.rb

```diff
@@ -5,10 +5,19 @@ class Space < ActiveRecord::Base
   has_many :trainings, dependent: :destroy
   has_many :training_sessions, through: :trainings
   has_many :certifications, through: :training_sessions
+
   validates :name,  presence: { message: "A name is required for the space"}, uniqueness: { message: "Space already exists"}
 
   def signed_in_users
-    return self.lab_sessions.where("sign_out_time > ?", Time.now).map(&:user)
+    return self.lab_sessions.where("sign_out_time > ?", Time.now).map(&:user).uniq
+  end
+
+  def recently_signed_out_users
+    users = self.lab_sessions.where(["(sign_out_time < ?) AND (sign_out_time > ?)", Time.now, 1.week.ago]).limit(40).map(&:user).uniq
```

Comment:
You probably want to order these by sign out time

Reply:
done